mypy/semanal.py

```diff
@@ -2686,6 +2686,8 @@ def check_and_set_up_type_alias(self, s: AssignmentStmt) -> bool:
 
         res: Optional[Type] = None
         if self.is_none_alias(rvalue):
+            # This is really `Type[None]`, not just `None`, but we use
+            # this as a backwards compatible solution. See #11550
             res = NoneType()
```

Comment:
This can still cause issues, for example:

```python
none = type(None)

print(none())  # E: error: Cannot instantiate type "Type[None]"
```

But, I've decided to keep this as-is (and do not change this to `Type[None]`), because I'm affraid this can cause issues to our old users, who use `type(None)` for annotations.

Reply:
If we decide to change this, we can try to make `Type[None]` and `None` compatible. So, this will work:

```python
n = type(None)
var: n = None  # will still be ok!
```

Reply:
I think types that are none should be annotated `None`. If someone tries to annotate something that is none as `type(None)` we should give an error and tell them to annotate it as `: None` (though I also expect this to be rather rare).